src/crab/ebpf_domain.cpp

```diff
@@ -1334,6 +1372,42 @@ void ebpf_domain_t::assign_valid_ptr(const Reg& dst_reg, bool maybe_null) {
     m_inv += reg.value <= PTR_MAX;
 }
 
+// If nothing is known of the stack_numeric_size,
+// try to recompute the stack_numeric_size.
+void ebpf_domain_t::recompute_stack_numeric_size(NumAbsDomain& inv, variable_t type_variable) {
+    variable_t stack_numeric_size_variable = variable_t::kind_var(data_kind_t::stack_numeric_sizes, type_variable);
+
+    if (!inv.eval_interval(stack_numeric_size_variable).is_top())
+        return;
+
+    if (type_inv.has_type(inv, type_variable, T_STACK)) {
+        int numeric_size =
+            stack.min_all_num_size(inv, variable_t::kind_var(data_kind_t::stack_offsets, type_variable));
+        if (numeric_size > 0)
+            inv.assign(stack_numeric_size_variable, numeric_size);
+    }
+}
+
+void ebpf_domain_t::recompute_stack_numeric_size(NumAbsDomain& inv, const Reg& reg) {
+    recompute_stack_numeric_size(inv, reg_pack(reg).type);
+}
+
+void ebpf_domain_t::add(const Reg& reg, int imm) {
+    auto dst = reg_pack(reg);
+    auto offset = get_type_offset_variable(reg);
+    add_overflow(dst.value, imm);
+    if (offset.has_value()) {
+        add(offset.value(), imm);
+        if (imm > 0)
+            // Since the start offset is increasing but
+            // the end offset is not, the numeric size decreases.
+            sub(dst.stack_numeric_size, imm);
```

Comment:
(Just a TODO for future PR):

The warning given by codecov led me to notice that here is the only case where stack_numeric_size is not TOP when `recompute_stack_numeric_size()` is called, and I'm not sure it is particularly needed. If recompute_stack_numeric_size() is moved inside `if (imm < 0)` then it can do the havoc() internally - or anything more sophisticated that might be possible.

Reply:
For code coverage, I added a test for this case, but the test just reproduces the issue in #321